JOSS submission (#502)
ivadomed/ivadomed: paper.md

## Loader

XX

## Training

XX

## Evaluation

XX

# Usage

XX # Mention (if applicable) a representative set of past or ongoing research projects using the software and recent scholarly publications enabled by it.

# Citations

Citations to entries in paper.bib should be in
[rMarkdown](http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html)
format.

If you want to cite a software repository URL (e.g. something on GitHub without a preferred
citation) then you can do it with the example BibTeX entry below for @fidgit.

For a quick reference, the following citation commands can be used:
- `@author:2001`  ->  "Author et al. (2001)"
- `[@author:2001]` -> "(Author et al., 2001)"
- `[@author1:2001; @author2:2001]` -> "(Author1 et al., 2001; Author2 et al., 2002)"

# Figures

Figures can be included like this:
![Caption for example figure.\label{fig:example}](figure.png)
and referenced from text using \autoref{fig:example}.

# Acknowledgements

> The authors thank Alexandru Jora, Nick Guenther, Christian Perone, Valentine Louis-Lucas, Benoît Sauty-De-Chalon, Alexandru Foias, Marie-Hélène Bourget and Leander Van Eekelen for their useful contributions, and Guillaume Dumas for proof-reading the manuscript. Funded by the Institut de valorisation des données (IVADO), the Canada Research Chair in Quantitative Magnetic Resonance Imaging [950-230815], the Canadian Institute of Health Research [CIHR FDN-143263], the Canada Foundation for Innovation [32454, 34824], the Fonds de Recherche du Québec - Santé [28826], the Natural Sciences and Engineering Research Council of Canada [RGPIN-2019-07244]. FRQNT Strategic Clusters Program (2020‐RS4‐265502 ‐ Centre UNIQUE  ‐Union Neurosciences \& Artificial Intelligence –Quebec, Canada First Research Excellence Fund through the TransMedTech Institute. C.G has a fellowship from IVADOMED [EX-2018-4], A.L. has a fellowship from NSERC and FRQNT, O.V. has a fellowship from NSERC, FRQNT and UNIQUE.

Reviewer: > C.G has a fellowship from IVADOMED

One day `ivadomed` will be able to provide fellowships 😉

---

# Summary

`ivadomed` is an open-source Python package for designing, end-to-end training, and evaluating deep learning models applied to medical imaging data. The package includes APIs, command-line tools, documentation, and tutorials. `ivadomed` also includes pre-trained models such as spinal tumor segmentation and vertebral labeling. Original features of `ivadomed` include a data loader that can parse image and subject metadata for custom data splitting or extra information during training and evaluation. Any dataset following the [Brain Imaging Data Structure (BIDS)](https://bids.neuroimaging.io/) convention will be compatible with `ivadomed`. Beyond the traditional deep learning methods, `ivadomed` features cutting-edge architectures, such as FiLM [@perez2017film] and HeMis [@havaei2016hemis], as well as various uncertainty estimation methods (aleatoric and epistemic), and losses adapted to imbalanced classes and non-binary predictions. Example applications of `ivadomed` include MRI object detection, segmentation, and labeling of anatomical and pathological structures. `ivadomed`'s main project page is available at https://ivadomed.org.

# Statement of need

> Deep learning applied to medical imaging presents many challenges: datasets are often not publicly-available, ground truth labels are difficult to produce, and needs are usually tailored to particular datasets and tasks [@kim_deep_2019]. There already exists a few deep learning frameworks for medical imaging, but [each have their pros & cons](https://ivadomed.org/en/latest/purpose.html#comparison-with-other-projects). `ivadomed` notably address unmet needs in terms of data management, readily-available uncertainty outputs, missing channels (in case of multi-channel training) and model comparison, to only name few of the original features.

Author: ```suggestion
Deep learning applied to medical imaging presents many challenges: datasets are often not publicly-available, ground truth labels are difficult to produce, and needs are usually tailored to particular datasets and tasks [@kim_deep_2019]. There already exists a few deep learning frameworks for medical imaging, but [each have their pros & cons](https://ivadomed.org/en/latest/purpose.html#comparison-with-other-projects). `ivadomed` notably address unmet needs in terms of data management, readily-available uncertainty outputs, missing modalities (in case of multi-channel training) and model comparison, to only name few of the original features.
```

> Deep learning applied to medical imaging presents many challenges: datasets are often not publicly-available, ground truth labels are difficult to produce, and needs are usually tailored to particular datasets and tasks [@kim_deep_2019]. There already exists a few deep learning frameworks for medical imaging, but [each have their pros & cons](https://ivadomed.org/en/latest/purpose.html#comparison-with-other-projects). `ivadomed` notably addresses unmet needs in terms of data management, readily-available uncertainty outputs, missing modalities (in case of multi-channel training) and model comparison, to only name few of the original features.

Reviewer 2: ```suggestion
Deep learning applied to medical imaging presents many challenges: datasets are often not publicly-available, ground truth labels are difficult to produce, and needs are usually tailored to particular datasets and tasks [@kim_deep_2019]. There already exists a few deep learning frameworks for medical imaging, but [each has their pros & cons](https://ivadomed.org/en/latest/purpose.html#comparison-with-other-projects). `ivadomed` notably addresses unmet needs in terms of data management, readily-available uncertainty outputs, missing modalities (in case of multi-channel training) and model comparison, to only name few of the original features.
```

> Another challenge of medical imaging is the heterogeneity of the data across hospitals (contrast, resolution, etc.), making it difficult to create models that can generalize well. Recent cutting-edge methods address this problem, such as FiLM [@perez2017film] and HeMis [@havaei2016hemis], however they are usually not implemented within a comprehensive framework. In addition to providing these architectures, `ivadomed` features losses adapted to imbalanced classes and non-binary predictions.

Author: ```suggestion
Another challenge of medical imaging is the heterogeneity of the data across hospitals (e.g., contrast, resolution), making it difficult to create models that can generalize well. Recent cutting-edge methods address this problem, such as FiLM [@perez2017film] and HeMis [@havaei2016hemis], however they are usually not implemented within a comprehensive framework. In addition to providing these architectures, `ivadomed` features losses adapted to imbalanced classes and non-binary predictions.
```

> ![Overview of `ivadomed` main features.\label{fig:overview}](https://raw.githubusercontent.com/ivadomed/doc-figures/main/overview.png)

Reviewer: now that we are no more restricted by space (but by words), i suggest we put back ivadomed's logo in the overview figure.

Reviewer: done in https://github.com/ivadomed/ivadomed/pull/502/commits/fcb1d87d5ee784c8d39263c4929c37f3202e89c5

> ![Overview of `ivadomed` main features.\label{fig:overview}](https://github.com/ivadomed/ivadomed/raw/master/images/overview.png)

Reviewer: update URL, see https://github.com/ivadomed/ivadomed/issues/511

Reviewer: done in b09a6e495a8f294050cc27ae41b2735f55b22a18

**Standard data structure:** In machine learning, lots of time is spent curating the data (renaming, filtering per feature) before entering the training pipeline [@Willemink2020-au]. `ivadomed` features an advanced data loader compatible with a standardized data structure in neuroimaging: the Brain Imaging Data Structure (BIDS) [@bids_2016]. Thus, any dataset following the BIDS convention can readily be used by `ivadomed`. BIDS convention is originally designed around MRI data and accepts NIfTI file formats, but the BIDS community is actively expanding its specifications to other modalities (CT, MEG/EEG, microscopy), which `ivadomed` will then be able to accommodate.

> **Access to metadata:** One benefit of the BIDS convention is that each image file is associated with a JSON file containing metadata. `ivadomed`'s loader can parse image metadata (e.g., acquisition parameters, image contrast, resolution) and subject metadata (e.g., pathology, age, sex) for custom data splitting or extra information during training and evaluation. It is possible to modulate specific layers of a convolutional neural network using metadata information to tailor it towards a particular data domain or to enable experiments with architectures such as FiLM [@perez2017film] (which is implemented in `ivadomed`). Metadata could also be useful to mitigate class imbalance via data balancing techniques.

Reviewer 2: ```suggestion
**Access to metadata:** One benefit of the BIDS convention is that each image file is associated with a JSON file containing metadata. `ivadomed`'s loader can parse image metadata (e.g., acquisition parameters, image contrast, resolution) and subject metadata (e.g., pathology, age, sex) for custom data splitting or extra information during training and evaluation. It is possible to modulate specific layers of a convolutional neural network using metadata information to tailor it towards a particular data domain or to enable experiments with architectures such as FiLM [@perez2017film]. Metadata could also be useful to mitigate class imbalance via data balancing techniques.
```
Pretty implicit + it is mention 3 or 4 times in the article that ivadomed has FiLM

## Training

**Architectures:** `ivadomed` includes all the necessary components for training segmentation models from start to finish, including data augmentation transformations and transfer learning. Available architectures include: 2D U-Net [@Ronneberger2015unet], 3D U-Net [@isensee2017brain], ResNet [@he2016resnet], DenseNet [@Huang2017densenet], Count-ception [@Cohen2017countception], and HeMIS U-Net. These models can easily be enriched via attention blocks [@oktay2018attention] or FiLM layers (which modulate U-Net features using metadata).

> **Losses and class imbalance:** Popular losses are available: Dice coefficient [@milletari2016v], cross-entropy, and L2 norm, including some adapted to medical imaging challenges, such as the adaptive wing loss [@wang_adaptive_2019] for soft labels and the focal Dice loss [@wong20183d] for class imbalance. Useful with small datasets, mixup [@zhang2017mixup] was modified to handle segmentation tasks. To mitigate class imbalance, `ivadomed` supports cascaded architectures. With a single inference, it is possible to narrow down the region of interest via object detection, and then segment a specific structure, as illustrated in \autoref{fig:lemay2020}.

Reviewer 2: ```suggestion
**Losses and class imbalance:** Popular losses are available: Dice coefficient [@milletari2016v], cross-entropy, and L2 norm, including some adapted to medical imaging challenges, such as the adaptive wing loss [@wang_adaptive_2019] for soft labels and the focal Dice loss [@wong20183d] for class imbalance. Useful to alleviate overfitting, mixup [@zhang2017mixup] was modified to handle segmentation tasks. To mitigate class imbalance, `ivadomed` supports cascaded architectures. With a single inference, it is possible to narrow down the region of interest via object detection and then segment a specific structure, as illustrated in \autoref{fig:lemay2020}.
```
Why is it useful on small datasets?

**Getting started:** It can be overwhelming to get started and choose across all the available models, losses, and parameters. `ivadomed`'s repository includes the script `ivadomed_automate_training` to configure and launch multiple trainings across GPUs. In case of interruption during training, all parameters are saved after each epoch so that training can be resumed at any time.

## Evaluation

**Uncertainty:** Aleatoric [@wang_aleatoric_2019] and/or epistemic [@nair_exploring_2018] uncertainties can be computed voxel-wise and/or object-wise [@roy_quicknat_2018]. Multiple metrics are available, including entropy and coefficient of variation.

> **Post-processing:** Predictions can be conveniently filtered using popular methods, e.g. fill holes, remove small objects, threshold using uncertainty. It is also possible to compute metrics for specific object sizes (e.g., small vs. large lesions). `ivadomed` has a module to find the optimal threshold value on the output soft prediction, via a grid-search finding applied to evaluation metrics or ROC curve.

Reviewer 2: ```suggestion
**Post-processing:** Predictions can be conveniently filtered using popular methods, e.g., fill holes, remove small objects, threshold using uncertainty. It is also possible to compute metrics for specific object sizes (e.g., small vs. large lesions). `ivadomed` has a module to find the optimal threshold value on the soft output prediction, via a grid-search finding applied to evaluation metrics or ROC curve.
```
or simply `soft prediction`

**Visualize performance:** Convenient visualization tools are available for model design and optimization: GIF animations across training epochs, visual quality control of data augmentation, training curve plots, integration of the TensorBoard module, and output images with true/false positive labels. See [this example tutorial](https://ivadomed.org/en/latest/tutorials/cascaded_architecture.html#visualize-training-data).

# Usage

> Past and ongoing research projects using `ivadomed` are listed [here](https://github.com/ivadomed/ivadomed/docs/source/use_cases.rst). The figure below illustrates a cascaded architecture for segmenting spinal tumors on MRI data [@lemay_fully_2020].

Reviewer: we need to redirect to RTD

> Past and ongoing research projects using `ivadomed` are listed [here](https://github.com/ivadomed/ivadomed/docs/source/use_cases.rst). The figure below illustrates a cascaded architecture for segmenting spinal tumors on MRI data [@lemay_fully_2020].

Reviewer: ```suggestion
Past and ongoing research projects using `ivadomed` are listed [here](https://ivadomed.org/en/stable/use_cases.html). The figure below illustrates a cascaded architecture for segmenting spinal tumors on MRI data [@lemay_fully_2020].
```

---
title: 'ivadomed: A Medical Imaging Deep Learning Toolbox'
tags:
  - Deep Learning
  - Medical Imaging
  - Segmentation
  - Open-source
  - Pipeline
authors:
  - name: Charley Gros
    orcid: 0000-0003-4318-0024
    affiliation: 1
  - name: Andreanne Lemay
    orcid: XX
    affiliation: 1
  - name: Olivier Vincent
    orcid: XX
    affiliation: 1
  - name: Lucas Rouhier
    orcid: XX
    affiliation: 1
  - name: Anthime Bucquet
    orcid: XX
    affiliation: 1
  - name: Joseph Paul Cohen
    orcid: XX
    affiliation: "2, 3"
  - name: Julien Cohen-Adad
    orcid: 0000-0003-3662-9532
    affiliation: "1, 4"
affiliations:
 - name: NeuroPoly Lab, Institute of Biomedical Engineering, Polytechnique Montreal, Montreal, Canada
   index: 1
 - name: AIMI, Stanford University, Stanford, CA, USA
   index: 2
 - name: Mila, Quebec AI Institute, Montreal, QC, Canada
   index: 3
 - name:  Functional Neuroimaging Unit, CRIUGM, Université de Montréal, Montreal, QC, Canada
   index: 4
date: 3 November 2020
bibliography: paper.bib

---

# Summary

> `ivadomed` is an open-source Python package for designing, end-to-end training, and evaluating deep learning models applied to medical imaging data. The package includes APIs, command-line tools, documentation, and tutorials. `ivadomed` also includes pre-trained models such as spinal tumor segmentation and vertebral labeling. Original features of `ivadomed` include a data loader that can parse image metadata (e.g., acquisition parameters, image contrast, resolution) and subject metadata (e.g., pathology, age, sex) for custom data splitting or extra information during training and evaluation. Any dataset following the [Brain Imaging Data Structure (BIDS)](https://bids.neuroimaging.io/) convention will be compatible with `ivadomed` without the need to manually organize the data, which is typically a tedious task. Beyond the traditional deep learning methods, `ivadomed` features cutting-edge architectures, such as FiLM [@perez2017film] and HeMis [@havaei2016hemis], as well as various uncertainty estimation methods (aleatoric and epistemic), and losses adapted to imbalanced classes and non-binary predictions. Each step is conveniently configurable via a single file. At the same time, the code is highly modular to allow addition/modification of an architecture or pre/post-processing steps. Example applications of `ivadomed` include MRI object detection, segmentation, and labeling of anatomical and pathological structures. Overall, `ivadomed` enables easy and quick exploration of the latest advances in deep learning for medical imaging applications. `ivadomed`'s main project page is available at https://ivadomed.org.

Reviewer: Removing some meat already present in subsequent sections
```suggestion
`ivadomed` is an open-source Python package for designing, end-to-end training, and evaluating deep learning models applied to medical imaging data. The package includes APIs, command-line tools, documentation, and tutorials. `ivadomed` also includes pre-trained models such as spinal tumor segmentation and vertebral labeling. Original features of `ivadomed` include a data loader that can parse image and subject metadata for custom data splitting or extra information during training and evaluation. Any dataset following the [Brain Imaging Data Structure (BIDS)](https://bids.neuroimaging.io/) convention will be compatible with `ivadomed` without the need to manually organize the data. Beyond the traditional deep learning methods, `ivadomed` features cutting-edge architectures, such as FiLM [@perez2017film] and HeMis [@havaei2016hemis], as well as various uncertainty estimation methods (aleatoric and epistemic), and losses adapted to imbalanced classes and non-binary predictions. Example applications of `ivadomed` include MRI object detection, segmentation, and labeling of anatomical and pathological structures. `ivadomed`'s main project page is available at https://ivadomed.org.
```

# Statement of need

> Deep learning is increasingly used in medical image processing [@kim_deep_2019]. It provides automated solutions to repetitive and/or tedious tasks such as the segmentation of pathological structures. However, medical imaging data present many challenges: datasets are often not publicly-available, ground truth labels are scarce due to the limited availability of expert raters, and needs can be very specific and tailored to particular datasets (e.g., segmentation of spinal tumors on sagittal MRI T2-weighted scans). Thus, offering solution for convenient training models (or fine-tuning of pre-existing models) is needed.

Reviewer: ```suggestion
Deep learning is increasingly used in medical image processing [@kim_deep_2019]. It provides automated solutions to repetitive and/or tedious tasks such as the segmentation of pathological structures. However, medical imaging data present many challenges: datasets are often not publicly-available, ground truth labels are scarce due to the limited availability of expert raters, and needs can be very specific and tailored to particular datasets and tasks. There already exists a variety of deep learning framework for medical imaging, but [each have their pros & cons](https://ivadomed.org/en/latest/purpose.html#comparison-with-other-projects). 
```

Author: I am a bit hesitant about integrating this one: we would be highlighting that (unfortunately) we are not the only one, without developing our strengths.

Reviewer: I think we need to mention it. At this point, there are already >15 similar toolboxes and ignoring them would be dishonest and would give the impression that we developed this toolbox in our bubble. If readers get that impression, this they will judge ivadomed very negatively. If we demonstrate we are aware of the state-of-the-art, it will give the impression that we can "only do better", because we know what's out there.  I'll find a rephrasing to spin ivadomed's contribution more positively.

Author: Mmmm I agree! THanks for explaining.
I agree even more when reading the [JOSS checklist](https://joss.readthedocs.io/en/latest/review_checklist.html#software-paper):
```
State of the field: Do the authors describe how this software compares to other commonly-used packages?
```

Another challenge of medical imaging is the heterogeneity of the data across clinical centers, in terms of image features (e.g., contrast, resolution) and population demographics. This makes it challenging to create models that can generalize well across the many existing datasets. Recent cutting-edge methods address this problem, such as FiLM [@perez2017film] and HeMis [@havaei2016hemis], however they are usually not implemented within a comprehensive framework that enables end-to-end training and experimentation. In addition to providing these advanced architectures, `ivadomed` features multiple uncertainty estimation methods (aleatoric and epistemic), losses adapted to imbalanced classes and non-binary predictions. Each step can be conveniently customized via a single configuration file, and at the same time, the code is highly modular to allow addition/modification of architecture or pre-/post-processing steps.

> # Software description
> 
> `ivadomed` is based on PyTorch framework [@paszke2017automatic] with GPU acceleration supported by CUDA. It can easily be installed via [PyPI](https://pypi.org/project/ivadomed/) and the whole package is tested with a [continuous integration](https://github.com/ivadomed/ivadomed/actions?query=workflow\%3A\%22Run+tests\%22) framework. The project website, which includes user and API documentation, is available at https://ivadomed.org. The name `ivadomed` is a portmanteau between [*IVADO (The Institute for data valorization)*](https://ivado.ca) and *medical*.
> 

Reviewer: I would remove because: 
- this section is optional
- the info is "somewhat" generic and can easily be found on the website
- we need to space

```suggestion
```

![Overview of `ivadomed` main features.\label{fig:overview}](https://github.com/ivadomed/ivadomed/raw/master/images/overview.png)

## Loader

> An important aspect of machine learning is data management. Lots of time is usually spent manually organizing data into a proper structure to make the dataset compatible with a chosen analysis pipeline [@Willemink2020-au]. `ivadomed` features a data loader module that expects datasets to be structured according to a widely-accepted convention: the Brain Imaging Data Structure (BIDS) [@bids_2016]. Thus, any dataset following the BIDS convention can immediately be used by `ivadomed`, e.g., for training a new model, without the need to spend time organizing the data. BIDS convention is designed around neuroimaging MRI data and accepts NIfTI file formats, but the BIDS community is actively expanding its specifications to other modalities (CT, MEG/EEG, microscopy) and file formats (PNG, OME-TIFF), which `ivadomed` will then be able to accommodate. 

Reviewer: removed some meat
```suggestion
In machine learning, lots of time is spent curating the data (renaming, filtering per features) before entering the training pipeline [@Willemink2020-au]. `ivadomed` features an advanced data loader compatible with a standardized data structure in neuroimaging: the Brain Imaging Data Structure (BIDS) [@bids_2016]. Thus, any dataset following the BIDS convention can readily be used by `ivadomed`. BIDS convention is originally designed around MRI data and accepts NIfTI file formats, but the BIDS community is actively expanding its specifications to other modalities (CT, MEG/EEG, microscopy), which `ivadomed` will then be able to accommodate. 
```

`ivadomed`'s data loader can accommodate 2D/3D images, multiple input channels as well as multiple prediction classes. Images can be loaded as a volume, slice-wise, or patch-wise. Data can be saved on the RAM or used "on the fly" via HDF5 format. Cropping, resampling, normalization, and histogram clipping and equalization can be applied during the loading as a pre-processing step. `ivadomed` can deal with missing modalities [@havaei2016hemis] by resorting to curriculum learning to train the model. 

## Training

> `ivadomed` includes all the necessary components for training segmentation models from start to finish. The first step usually consists of applying data augmentation transformations, such as affine, elastic, and ground-truth dilation, which are all included in `ivadomed`. The next step is model training, which can be done either from scratch or using transfer learning on a pre-trained model by freezing some layers. Available architectures include: 2D U-Net [@Ronneberger2015unet], 3D U-Net [@isensee2017brain], ResNet [@he2016resnet], DenseNet [@Huang2017densenet], Count-ception [@Cohen2017countception], and HeMIS U-Net. These models can easily be enriched via attention blocks [@oktay2018attention] or FiLM layers (which modulate U-Net features using metadata). To facilitate the training process, `ivadomed` offers multiple loss functions such as the Dice coefficient [@milletari2016v], cross-entropy, and L2 norm, including some adapted to medical imaging challenges, such as the adaptive wing loss [@wang_adaptive_2019] for soft labels and the focal Dice loss [@wong20183d] for class imbalance. To partly address the problem of small datasets, mixup [@zhang2017mixup] has been implemented and adapted for segmentation tasks. To mitigate class imbalance, `ivadomed` supports cascaded architectures. With a single inference, it is possible to narrow down the region of interest via object detection, and then segment a specific structure. In case of interruption during training, all parameters are saved after each epoch so that training can be resumed at any time. 

Reviewer: Removed a bit of meat, added ref to Figure 2 (tumor seg)
```suggestion
`ivadomed` includes all the necessary components for training segmentation models from start to finish, including data augmentation transformations and transfer learning. Available architectures include: 2D U-Net [@Ronneberger2015unet], 3D U-Net [@isensee2017brain], ResNet [@he2016resnet], DenseNet [@Huang2017densenet], Count-ception [@Cohen2017countception], and HeMIS U-Net. These models can easily be enriched via attention blocks [@oktay2018attention] or FiLM layers (which modulate U-Net features using metadata). To facilitate the training process, `ivadomed` offers multiple loss functions such as the Dice coefficient [@milletari2016v], cross-entropy, and L2 norm, including some adapted to medical imaging challenges, such as the adaptive wing loss [@wang_adaptive_2019] for soft labels and the focal Dice loss [@wong20183d] for class imbalance. To partly address the problem of small datasets, mixup [@zhang2017mixup] has been modified to handle segmentation tasks. To mitigate class imbalance, `ivadomed` supports cascaded architectures. With a single inference, it is possible to narrow down the region of interest via object detection, and then segment a specific structure, as shown in Figure 2. In case of interruption during training, all parameters are saved after each epoch so that training can be resumed at any time. 
```

> It can be overwhelming to get started and choose across all the available models, losses, and parameters. `ivadomed`'s repository includes the script `ivadomed_automate_training` to configure and launch multiple trainings across GPUs. In addition, `ivadomed` includes tutorials that cover typical training scenarios such as one-class segmentation, cascade of architectures, and uncertainty estimation. 

Reviewer: If we need space i think it can be removed
```suggestion
```

Author: `ivadomed_automate_training`: This is quite a cool feature that we have not seen in other repo'. But I agree to remove the tutorial sentence!

Multiple visualization tools are included in `ivadomed` to support the design and optimization of tailored training models: GIF animations across training epochs, visual quality control of data augmentation, training curve plots, integration of the TensorBoard module, and output images with true/false positive labels. 

# Usage

> Past or ongoing research projects using `ivadomed` are listed [here](https://github.com/ivadomed/ivadomed/docs/source/use_cases.rst).

Reviewer: i think it would be *really* cool to add Andréanne's figure about tumor segmentation. People *love* figures.

Author: Yeah!! Done in [608a24d](https://github.com/ivadomed/ivadomed/pull/502/commits/608a24d9044de12cab84eb12acb37be4dacdcb5d).

> Past or ongoing research projects using `ivadomed` are listed [here](https://github.com/ivadomed/ivadomed/docs/source/use_cases.rst).

Reviewer: will need to be replaced by the built html URL on RTD
